main/src/ca/uwaterloo/flix/language/phase/typer/TypeReduction.scala

```diff
@@ -258,21 +278,54 @@ object TypeReduction {
     val static = java.lang.reflect.Modifier.isStatic(cand.getModifiers)
-    (if (isStatic) static else !static) &&
-      (cand.getName == methodName) &&
-      (cand.getParameterCount == ts.length) &&
-      // Parameter types correspondence with subtyping
-      (cand.getParameterTypes zip ts).forall {
-        case (clazz, tpe) => isSubtype(tpe, Type.getFlixType(clazz))
-      } &&
-      // NB: once methods with same signatures have been filtered out, we should remove super-methods duplicates
-      // if the superclass is abstract or ignore if it is a primitive type or void
-      (cand.getReturnType.equals(Void.TYPE) || cand.getReturnType.isPrimitive || cand.getReturnType.isArray || // for all arrays return types?
-        java.lang.reflect.Modifier.isInterface(cand.getReturnType.getModifiers) || // interfaces are considered primitives
-        !(java.lang.reflect.Modifier.isAbstract(cand.getReturnType.getModifiers) && !isStatic))
-  } // temporary to avoid superclass abstract duplicate, except for static methods
+    if (isStatic != static) return false
+    if (!cand.getName.equals(methodName)) return false
+    // Check candidate depending on if it is variadic
+    if (isVariadic) {
+      if (!isCandidateVariadicMethod(cand, ts)) return false
+    } else {
+      val validSubtyping =
+        (cand.getParameterCount == ts.length) &&
+        // Parameter types correspondence with subtyping
+        (cand.getParameterTypes zip ts).forall {
+          case (clazz, tpe) => isSubtype(tpe, Type.getFlixType(clazz))
+        }
+      if (!validSubtyping) return false
+    }
+    // NB: once methods with same signatures have been filtered out, we should remove super-methods duplicates
+    // if the superclass is abstract or ignore if it is a primitive type or void
+    val duplicatesFilter = (cand.getReturnType.equals(Void.TYPE) || cand.getReturnType.isPrimitive || cand.getReturnType.isArray || // for all arrays return types?
+      java.lang.reflect.Modifier.isInterface(cand.getReturnType.getModifiers) || // interfaces are considered primitives
+      !(java.lang.reflect.Modifier.isAbstract(cand.getReturnType.getModifiers) && !isStatic)) // temporary to avoid superclass abstract duplicate, except for static methods
+
+    duplicatesFilter
+  }
+
+  /**
+   * Helper method used in isCandidateMethod to check if the given method corresponds to a variadic method given
+   * an arguments list.
+   */
+  private def isCandidateVariadicMethod(cand: Method, ts: List[Type])(implicit flix: Flix): Boolean = {
+    if (!cand.isVarArgs)
+      return false
+    val paramCount = cand.getParameterCount
+    val varArr = cand.getParameterTypes.last
+    val tpe = varArr.getComponentType
+    // Check minimum required arity
+    // e.g., for a defined function f(T arg1, T... args) arg1 is required.
+    val argsDiff = paramCount - ts.length
+    if (argsDiff > 1) // We remove the vararg array and check for required arguments
```

Comment:
you write that you now check something, but it just returns false, which makes the other method return false. is the comment outdated?